### ISSUE TITLE
Integer columns with missing data are always displayed as precision 1 float

### ISSUE THREAD

**Issue by kilpert** [datavzrd/datavzrd#373]
When there is missing data in an integer column, it is always rendered as a precision 1 float. This cannot be mitigated, e.g. by setting precision to 0 for that column.

**Comment by fxwiegand** [comment]
Could you please provide a minimal example? I cannot recreate this with our example report: 
<img width="118" alt="Bildschirmfoto 2023-02-21 um 13 18 07" src="https://user-images.githubusercontent.com/39430842/220343050-f85b1ad1-9eef-4746-8381-eb8583b81521.png">

**Comment by stefanpeidli** [comment]
Hi and thanks for your awesome tool!

I have a similar issue, where I see different precisions despite me setting them all to zero.
In my case it looks like this:
![Screenshot 2023-12-05 at 16 26 39](https://github.com/datavzrd/datavzrd/assets/24811160/e7a2d914-4c10-4318-a58d-3258106fb64c)


**Minimal example** (with datavzrd 2.33.0):
Using this [data.csv](https://github.com/datavzrd/datavzrd/files/13561966/data.csv) with two float columns where one of them has two empty entries, and this config:
```
name: precision test
datasets:
  dataset_info:
    path: "data.csv"
default-view: dataset_info
views:
  dataset_info:
    dataset: dataset_info
    page-size: 25
    render-table:
      columns:
        Column without Nans:
          precision: 0
        Column with Nans:
          precision: 0
```

Result:
![Screenshot 2023-12-05 at 16 25 02](https://github.com/datavzrd/datavzrd/assets/24811160/fb60a2ff-42ac-46c3-a20c-247c7b48c215)

**Comment by fxwiegand** [comment]
So from what I can see in your example the problem does not origin in the raw data containing NaN entries. My suspicion would much rather be that this is caused by the precision being set to 0? I will take a look at this asap! Thanks a lot for providing the minimal example!

**Comment by fxwiegand** [comment]
> Hi and thanks for your awesome tool!
> 
> I have a similar issue, where I see different precisions despite me setting them all to zero. In my case it looks like this: ![Screenshot 2023-12-05 at 16 26 39](https://private-user-images.githubusercontent.com/24811160/288093633-e7a2d914-4c10-4318-a58d-3258106fb64c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIwMzUwMTEsIm5iZiI6MTcwMjAzNDcxMSwicGF0aCI6Ii8yNDgxMTE2MC8yODgwOTM2MzMtZTdhMmQ5MTQtNGMxMC00MzE4LWE1OGQtMzI1ODEwNmZiNjRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA4VDExMjUxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM2N2E3ODViZWE1Mzc4YTgxMjE3NjViNzkwNTFlYzViOWI5YmFhZDU1MTk1M2NkMTY0ODNkZGQ4NTFlMjMzMWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.m0jZXf__2rLR-babfW48YM2DcL1lfu1vimxSabI-e_o)
> 
> **Minimal example** (with datavzrd 2.33.0): Using this [data.csv](https://github.com/datavzrd/datavzrd/files/13561966/data.csv) with two float columns where one of them has two empty entries, and this config:
> 
> ```
> name: precision test
> datasets:
>   dataset_info:
>     path: "data.csv"
> default-view: dataset_info
> views:
>   dataset_info:
>     dataset: dataset_info
>     page-size: 25
>     render-table:
>       columns:
>         Column without Nans:
>           precision: 0
>         Column with Nans:
>           precision: 0
> ```
> 
> Result: ![Screenshot 2023-12-05 at 16 25 02](https://private-user-images.githubusercontent.com/24811160/288092655-fb60a2ff-42ac-46c3-a20c-247c7b48c215.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIwMzUwMTEsIm5iZiI6MTcwMjAzNDcxMSwicGF0aCI6Ii8yNDgxMTE2MC8yODgwOTI2NTUtZmI2MGEyZmYtNDJhYy00NmMzLWEyMGMtMjQ3YzdiNDhjMjE1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA4VDExMjUxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIzMjA3N2U2ZGQyMDc2NzhjODEzZTc3YzBlYzZkZTExZjMzMDhiYjk1ODdhN2YyYzI1M2QzZDlhM2E0YzRhMWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0._pCIGYYLTlnID4NoIj6hovCupLYmAFgn9OlqvaN8R-M)

I think your problem should be solved when #596 is merged! Thanks again for reporting the problem especially with the minimal example!
